content/docs/rfcs/57/README.md

```diff
@@ -287,16 +287,88 @@ This allows the Status app to abstract from the discovery process and simply add
+
+This solution introduces two roles:
+* Publisher: A node that knows the `private-key-topic` associated to `public-key-topic`, that can publish messages with a valid `message-signature` that are accepted and relayed by the nodes implementing this feature.
+* Relayer: A node that knows the `public-key-topic`, which can be used to verify if the messages were signed with the equivalent `private-key-topic`. It allows distinguishing valid from invalid messages which protect the node against DoS attacks, assuming that the users of the key send messages of a reasonable size and rate. Note that a node can validate messages and relay them or not without knowing the private key.
+
+## Design requirements (publisher)
+
+A publisher that wants to send messages that are relayed in the network for a given `protected-pubsub-topic` shall:
+* be able to sign messages with the `private-key-topic` configured for that topic, producing a ECDSA signature of 64 bytes.
+* include the signature of the `app-message-hash` (`message-signature`) that wishes to send in the `WakuMessage` `meta` field.
+
+## Design requirements (relay)
+
+Requirements for the relay are listed below:
+
+* A valid `protected-pubsub-topic` shall be configured with a `public-key-topic`, (derived from a `private-key-topic`). Note that the relay does not need to know the private key.
+For simplicity, there is just one key per topic. Since this approach has clear privacy implications, this configuration is not part of the waku protocol, but of the application.
+* Relay nodes should leverage the existing gossipsub validators that allow to `Accept` or `Reject` messages.
+* Upon receiving a message, the node shall check the `meta` `WakuMessage` field. If empty, `Reject` the message.
+* If `meta` exists but its size is different than 64 bytes, `Reject` the message.
+* If `meta` exists and has a size of 64 bytes, assert that `message-signature` is verified according to the ECDSA signature verification algorithm using `public-key-topic` and `app-message-hash`.
+* If the signature does not verify correctly, `Reject` the message.
+* If and only if the signature is verified, `Accept` the message.
+* The node shall keep metrics on the messages validation output, `Accept` or `Reject`.
+* (Optional). To further strengthen DoS protection, gossipsub [scoring](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#extended-validators) can be used to trigger disconnections from peers sending multiple invalid messages. See `P4` penalty.
+This protects each peer from DoS, since this score is used to trigger disconnections from nodes attempting to DoS them.
+
+
+## Required changes
+
+This solution is designed to be backward compatible so that nodes validating messages can coexist in the same topic with other nodes that don't perform validation. But note that only nodes that perform message validation will be protected against DoS. Nodes wishing to opt-in this DoS protection feature shall:
```

Comment:
> But note that only nodes that perform message validation will be protected against DoS.

This is not entirely true as a relay node only connected to node that perform message validation are also protected.